### ISSUE TITLE
Documents for Highly Available Topology

### ISSUE THREAD

**Issue by RainbowMango** [karmada-io/website#418]
**What documentation would you like to be added**:

Add some documents about how to deploy the HA（highly available） Karmada.

**Why do you think this document is needed**:

I believe that Karmada is capable of high availability because I have heard that some users have already done so, but so far, there are no relevant docs in Karmada that describe how to deploy a high availability control plane.

Many users would need this document, such as :
- https://github.com/karmada-io/karmada/discussions/3773
- TBD

**Where do you think the document should be placed**:

- [ ] Highly Available Topology Overview
  - this document could be placed at https://github.com/karmada-io/website/tree/main/docs/installation/ha-installation.md
  - reference: [Kuberentes Options for Highly Available Topology](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/ha-topology/)
- [ ] Creating Highly Available Karmada with karmadactl
  - this document could be placed at https://github.com/karmada-io/website/tree/main/docs/installation/ha-installation-with-cli.md
  - reference: [Creating Highly Available Clusters with kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/)
- [ ] Creating Highly Available Karmada with helm chart
  - this document could be placed at https://github.com/karmada-io/website/tree/main/docs/installation/ha-installation-with-helm-chart.md
- [ ] Creating Highly Available Karmada with karmada-operator
  - this document could be placed at https://github.com/karmada-io/website/tree/main/docs/installation/ha-installation-with-operator.md
- [ ] Does Karmada support high availability(HA) deployments?
  - add an item to FAQ at https://github.com/karmada-io/website/blob/main/docs/faq/faq.md

**Comment by Affan-7** [comment]
I have successfully created a karmada cluster and a member cluster on my local machine using kind. I also registered the member cluster to the karmada. But when I try to deploy karmada with HA using `kubectl karmada init --karmada-apiserver-replicas 3 --etcd-replicas 3` command. It give me the following error.

`error: for data security,when etcd storage mode is hostPath,etcd-replicas can only be 1`

Is there a solution to create multiple etcd storages with hostPath storage mode?

**Comment by samzong** [comment]
Hi @Affan-7 

Thanks for your choose the Karmada.

I'm not an expert in this area. However, based on the error message, my suggestion is that you could try to resolve the problem by using a storageclass or an external etcd. 

- https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/setup-ha-etcd-with-kubeadm/

Using a single replica of etcd in a production environment is dangerous. 

If you plan to use Karmada in a production environment, please let us know at the earliest opportunity. 

If you have any more questions, feel free to create an Issue or join our official Slack channel at any time.

**Comment by Abhay349** [comment]
@RainbowMango Thanks for suggestion,
I’ll first spend some time understanding the existing HA-related discussions, control-plane architecture, and the current installation methods in Karmada. 

After that, I plan to share a short proposal for the “Highly Available Karmada with Helm” guide so that  we can confirm the structure and scope before I start writing the full document. **This will also help me deepen my understanding of Karmada’s control-plane design while contributing something useful for new users**.

Here is a brief draft outline for the guide. Please let me know if this direction makes sense:

1. Studying the existing HA topology documentation and control-plane architecture.
2. Set up a local multi-node environment using Kind or test clusters.
3. Deploy Karmada in HA mode using the Helm chart.
4. Verify:
   - Multiple replicas of control-plane components
   - Leader election behavior
   - Failover when a control-plane pod is stopped
5. Document:
   - Architecture overview
   - Step-by-step Helm installation
   - Verification steps
   - Common pitfalls and troubleshooting tips.

Thanks again :)

**Comment by oliviacaraiman** [comment]
Hi @RainbowMango, 
I’m trying to align my production HA design with Karmada guidance, and it seems to me that there is a gap between the official HA documentation and current implementation behavior.
In [docs](https://karmada.io/docs/next/installation/ha-installation/), HA is described with “stacked etcd topology” language that sounds like each etcd instance should be co-located with a server instance in the same node. But in current Helm manifests, internal etcd appears to run as a StatefulSet and control-plane components connect via etcd service/endpoints, not strict localhost co-location.
Could you please help clarify if it's a doc/implementation drift or am I missing something?

**Comment by RainbowMango** [comment]
> I’m trying to align my production HA design with Karmada guidance,

What is your production HA? Do you mean you are running Karmada in your production environment?

**Comment by oliviacaraiman** [comment]
> > I’m trying to align my production HA design with Karmada guidance,
> 
> What is your production HA? Do you mean you are running Karmada in your production environment?

What I meant is that I am currently running Karmada in a single replica mode for all components, and to make it Production ready, I want to set up HA for etcd and Control Plane. So I was reading the documentation and tried to update the helm templates, and it seems to me that the docs are not reflecting the set up you currently have in the charts https://github.com/karmada-io/karmada/blob/a957f64d50213729b4d34f3c298ca3630e1c9127/charts/karmada/values.yaml#L713. I don't see how are you ensuring that each api server is talking to a particular instance of etcd in internal mode (which is the definition of stacked topology). So, my question is:  is the documentation accurately describing the current implementation? If not, is it possible to update it?

**Comment by RainbowMango** [comment]
Yeah, I see. I will get back to this issue and confirm it as soon as possible.

By the way, are you evaluating Karmada right now? Would you mind telling us which commany you work for?
